src/components/Carousel/index.tsx

```diff
@@ -70,23 +70,19 @@ export default class Carousel extends React.Component<CarouselProps, CarouselSta
         renderItem: (item: React.ReactNode) => {
             return item;
         },
-        renderThumbs: (children: React.ReactChild[]) => {
-            const images = Children.map<React.ReactChild | undefined, React.ReactChild>(children, (item) => {
-                let img: React.ReactChild | undefined = item;
-
-                // if the item is not an image, try to find the first image in the item's children.
-                if ((item as React.ReactElement<{ children: React.ReactChild[] }>).type !== 'img') {
-                    img = (Children.toArray((item as React.ReactElement).props.children) as React.ReactChild[]).find(
-                        (children) => (children as React.ReactElement).type === 'img'
-                    );
-                }
+        renderThumbs: (children: React.ReactNode) => {
+            const findFirstImage = (children: React.ReactNode) =>
+                Children.toArray(children)
+                    .filter(React.isValidElement)
```

Comment:
Filtered on isValidElement to ensure we don't have non-element children (e.g. string | number). Otherwise TS wouldn't be sure that we actually have `type` property on the child object. 

This is more correct behaviour and would support the actual possibility of a string or number (totally acceptable react children) to enter this function, but does add some performance overhead given the extra filtering needed. Not sure how crucial that is for this use case?